File: mach_write.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <cstdlib>
#include <iostream>
#include <stdlib.h>
#include <string>
#include <thread>
#include <chrono>
#include <random>
#include <stdio.h>
using namespace std;


FILE *filestream;
string text;
char buffer[4096];
int minum, maxum;

random_device rd;
mt19937 gen(rd());


void  MachineWriting(string text)
{
    for(int i = 0; i <= text.length(); i++)
    {
        uniform_int_distribution<>normal(minum, maxum);
        if(text[i] == *".")
        {
            minum = 350;
            maxum = 450;

        }
        else if(text[i] == *"/")
        {
            break;
        }
        else 
        {
            minum = 90;
            maxum = 190;
        }
        this_thread::sleep_for(chrono::milliseconds(normal(gen)));
        cout<<text[i];
    }
    cout<<endl;
}

int main()
{
    setlocale(LC_ALL,"russian");
    filestream = fopen("text.txt","r");
    fread(buffer,sizeof buffer,1,filestream);
    MachineWriting(text.assign(buffer, buffer + sizeof buffer));
    system("pause");
    
}

File: MachWrite.cs
using System;
using System.Threading;
using System.IO;


class MachineWriting
{
	public static void MachType(string Text, int MinGen, int MaxGen)
	{
		Random randModule = new Random();
		for (int i = 0; i < Text.Length; i++)
		{
			Console.Write(Text[i]);
			switch (Text[i])
			{
				//если в тексте есть запятая, randModule генерирует число, на 150 пунктов больше чем изначально
				case ',':
				Thread.Sleep(randModule.Next(MinGen + 150,MaxGen + 150));
				break;
				//если в тексте есть точка, randModule генерирует число, которое в среднем на 300 пунктов больше
				case '.':
				Thread.Sleep(randModule.Next(MinGen + 300,MaxGen + 300));
				break;
				
				default:
				Thread.Sleep(randModule.Next(MinGen,MaxGen));
				break;
			}
		}
	}
}